File: code_for_interview/10.斐波那契数列.cpp
#include <iostream>
#include <stack>
#include <vector>

using namespace std;

class Solution {
public:
    int Fibonacci(int n) {
        // int a,b,c;
        int a = 0,b = 0,c = 0;
        for (int i = 1; i <= n; ++i) {
            if (i==1){
                c=1;
            } else if(i==2){
                b=0;
                a=1;
                c=1;
            } else {
                b = a;
                a = c;
                c = b + a;
            }
        }
        return c;
    }
};
int main() {
    Solution s;
    cout<<s.Fibonacci(1)<<endl;
    cout<<s.Fibonacci(2)<<endl;
    cout<<s.Fibonacci(3)<<endl;
    cout<<s.Fibonacci(5)<<endl;
    cout<<s.Fibonacci(7)<<endl;
    return 0;
}

File: code_for_interview_py/10.斐波那契数列.py
# -*- coding:utf-8 -*-
class Solution:
    def Fibonacci(self, n):
        # write code here
        if n == 0:
            return 0
        if n == 1:
            return 1
        pre1 = 0
        pre2 = 1
        for _ in range(2, n+1):
            cur = pre1+pre2
            pre1 = pre2
            pre2 = cur
        return cur